src/lib/get-commits.js

```diff
@@ -0,0 +1,57 @@
+const execa = require('execa');
+const log = require('npmlog');
+const SemanticReleaseError = require('@semantic-release/error');
+
+module.exports = async ({lastRelease, options}) => {
+  let stdout;
+  if (lastRelease.gitHead) {
+    try {
+      ({stdout} = await execa('git', ['branch', '--no-color', '--contains', lastRelease.gitHead]));
+    } catch (err) {
+      throw notInHistoryError(lastRelease.gitHead, options.branch);
+    }
+    const branches = stdout
+      .split('\n')
+      .map(branch => branch.replace('*', '').trim())
+      .filter(branch => !!branch);
+
+    if (!branches.includes(options.branch)) {
+      throw notInHistoryError(lastRelease.gitHead, options.branch, branches);
+    }
+  }
+
+  try {
+    ({stdout} = await execa('git', [
+      'log',
+      '-E',
+      '--format=%H==SPLIT==%B==END==',
+      `${lastRelease.gitHead ? lastRelease.gitHead + '..' : ''}HEAD`,
+    ]));
```

Comment:
do we need to wrap this block into `()`?

Reply:
I’ve seen it in a few other places, too

Reply:
Yes in order to reuse the `stdout` variable.
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration

I could also use different variables.

Reply:
ah got it, it’s fine, code is readable as is, I was just curious. And learned something, thanks 👍

Reply:
I'm thinking of doing a bit more refactoring in other PR actually. Mostly modularize a bit and splitting thing in different function.
That would make the code clearer, allow us to write more docs in the form of jsdoc.
Also that would simplify a lot the situation like were have to reuse variable.

Most of the case I do that is due to `try` / `catch` to avoid having all the code depending the throwing function be part of the try block.

With more modularization, each function that throws could handle it's errors (ie process the error and rethrow a SemanticRealseError) so in the parent module that calls them we could have only one `try` `catch` block.

Reply:
> I'm thinking of doing a bit more refactoring in other PR actually. Mostly modularize a bit and splitting thing in different function.
That would make the code clearer, allow us to write more docs in the form of jsdoc.
Also that would simplify a lot the situation like were have to reuse variable.

yeah +1 on doing that in follow up PRs, this one is quite big already

> With more modularization, each function that throws could handle it's errors (ie process the error and rethrow a SemanticRealseError) so in the parent module that calls them we could have only one try catch block.

Sounds good to me